### PR TITLE
doc: refactor readme files to better explain the usage of the provider

### PR DESCRIPTION
**Pull request by cristiGuranIonos**
## What does this fix or implement?

- Move migration details to the bottom.
- Move `using the provider` to the top.
- Suggest to use token auth. Link to generating a token.

## Checklist

- [ ] PR name added as appropriate (e.g. `feat:`/`fix:`/`doc:`/`test:`/`refactor:`)
- [ ] Tests added or updated
- [ ] Documentation updated
- [ ] Changelog updated and version incremented (label: upcoming release)
- [ ] Github Issue linked if any
- [ ] Jira task updated

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -173,4 +128,96 @@ now you can see the response body incl. api error message:
     "message" : "[VDC-yy-xxxx] Operation cannot be executed since this Kubernetes Nodepool is already marked for deletion. Current state of the resource is FAILED_DESTROYING."
   }]
 }
-```
\ No newline at end of file
+```
+
+## Building The Provider
```
Comment (mimihalescu) [inline README.md:133]: An idea: we could move this information about building, developing, testing in a different file that is relevant only for the contributors of the provider

File: docs/index.md [hunk 2]
```diff
@@ -287,3 +231,68 @@ Individual resources must opt-in to providing configurable `timeouts`, and attem
 ~> **Note:** Terraform does not automatically rollback in the face of errors.
 Instead, your Terraform state file will be partially updated with
 any resources that successfully completed.
+
+## Migrating from the ProfitBricks provider
```
Comment (mimihalescu) [inline docs/index.md:235]: Also, the migration guide can be moved to a different file, so only interested users (if there still are users who use profitbricks) would access it

**Review by mimihalescu — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -17,127 +17,82 @@
...
-```
-
-Enter the provider directory and build the provider
-
-```sh
-$ cd $GOPATH/src/github.com/ionos-cloud/terraform-provider-ionoscloud
-$ make build
-```
-
 ## Using the provider
```
Comment (cbeti-ionos) [inline README.md:32]: Section could be renamed to Provider Usage Reference and moved below

File: README.md (hunk 1 above)
Comment (cristiGuranIonos) [reply README.md:131]: good idea. will leave it only in the Readme in github

File: docs/index.md (hunk 2 above)
Comment (cristiGuranIonos) [reply docs/index.md:235]: I actually removed it, I don't think there is anyone left using the profitbricks provider.

**Pull request merged**